### PR TITLE
Make inputs visible while typing on mobile

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

This PR does two things: reduces margins for most pages (around signup/login where there are text inputs) so that you can see what you're typing while you type, and adds a use of React Native's new KeyboardAvoidingView for one page (the "PGP Add" page with full name and emails).  KeyboardAvoidingView works great!

Here's a before-and-after (before on the left) MOV showing the change on iOS during a signup, and then the PGP Add screen in the dumb view at the end:

http://chris.printf.net/before-and-after.mov

### CODE REVIEW

File: shared/common-adapters/progress-indicator.ios.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
 // @flow
 import React, {Component} from 'react'
 import type {Props} from './progress-indicator'
-import {ActivityIndicatorIOS} from 'react-native'
+import {ActivityIndicator} from 'react-native'
```
Comment (cjb) [inline shared/common-adapters/progress-indicator.ios.js]: this is an API change, the IOS version is deprecated/throws a warning

File: shared/login/register/set-public-name/index.render.native.js [hunk 2]
```diff
@@ -13,10 +13,11 @@ const SetPublicName = ({onBack, onSubmit, onChange, deviceNameError, deviceName,
         style={stylesContainer}
         onBack={onBack}>
         <Text type='Header' style={stylesHeader}>Set a public name for this device:</Text>
-        <Icon type='icon-phone-colors-64' style={stylesIcon} />
+        <Icon type='icon-phone-colors-32' style={stylesIcon} />
```
Comment (cjb) [inline shared/login/register/set-public-name/index.render.native.js]: zeplin uses the smaller icon size

File: shared/login/register/set-public-name/index.render.native.js [hunk 3]
```diff
@@ -13,10 +13,11 @@ const SetPublicName = ({onBack, onSubmit, onChange, deviceNameError, deviceName,
...
         <Input
           autoFocus={true}
           style={stylesInput}
+          errorText={deviceNameError}
```
Comment (cjb) [inline shared/login/register/set-public-name/index.render.native.js]: this was a bug -- errorText wasn't hooked up on the device name page on mobile.

File: shared/login/signup/invite-code.render.native.js [hunk 4]
```diff
@@ -25,41 +25,35 @@ class Render extends Component {
...
-    marginTop: 22,
-  },
-  input: {
-    alignSelf: 'stretch',
-    marginTop: 0,
-  },
-  text: {
-    marginTop: 32,
-  },
+const stylesContainer = {
```
Comment (cjb) [inline shared/login/signup/invite-code.render.native.js]: Tried to clean up some old-style styles and numeric margins while I was here.

File: shared/login/signup/username-email-form.render.native.js [hunk 5]
```diff
@@ -3,15 +3,15 @@ import Container from '../forms/container'
...
+import {globalColors, globalMargins} from '../../styles/style-guide'
 
 class Render extends Component<void, Props, void> {
   render () {
     return (
       <Container onBack={this.props.onBack} outerStyle={stylesOuter}>
         <UserCard>
-          <Input autoFocus={true} hintText='Create a username' value={this.props.username} errorText={this.props.usernameErrorText} onChangeText={username => this.props.usernameChange(username)} />
-          <Input style={stylesInput2} hintText='Email address' value={this.props.email} errorText={this.props.emailErrorText} onEnterKeyDown={this.props.onSubmit} onChangeText={email => this.props.emailChange(email)} keyboardType='email-address' />
+          <Input autoFocus={true} hintText='Create a username' floatingLabelText='Create a username' value={this.props.username} errorText={this.props.usernameErrorText} onChangeText={username => this.props.usernameChange(username)} />
```
Comment (chrisnojima) [inline shared/login/signup/username-email-form.render.native.js]: can you remove hintText if you add floating? i thought so...

**Comment by chrisnojima** [comment]
👍

File: shared/login/signup/username-email-form.render.native.js (hunk 5 above)
Comment (cjb) [inline shared/login/signup/username-email-form.render.native.js]: > can you remove hintText if you add floating? i thought so...

Only on desktop, not native.  Added a ticket to make the semantics match: DESKTOP-1773.

**Pull request merged**